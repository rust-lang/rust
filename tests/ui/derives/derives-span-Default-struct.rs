// This file was auto-generated using 'src/etc/generate-deriving-span-tests.py'


struct Error;

#[derive(Default)]
struct Struct {
    x: Error //~ ERROR
}

fn main() {}
