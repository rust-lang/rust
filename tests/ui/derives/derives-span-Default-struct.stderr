error[E0277]: the trait bound `Error: Default` is not satisfied
  --> $DIR/derives-span-Default-struct.rs:8:5
   |
LL | #[derive(Default)]
   |          ------- in this derive macro expansion
LL | struct Struct {
LL |     x: Error
   |     ^^^^^^^^ the trait `Default` is not implemented for `Error`
   |
   = note: this error originates in the derive macro `Default` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider annotating `Error` with `#[derive(Default)]`
   |
LL + #[derive(Default)]
LL | struct Error;
   |

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0277`.
